tests/Unit/ControlSystem/Actions/Test_InitializeMeasurements.cpp

```diff
@@ -0,0 +1,190 @@
+
+#include "ControlSystem/Actions/InitializeMeasurements.hpp"
+#include "ControlSystem/Event.hpp"
+#include "ControlSystem/Protocols/ControlSystem.hpp"
+#include "ControlSystem/Protocols/Measurement.hpp"
+#include "ControlSystem/Protocols/Submeasurement.hpp"
+#include "ControlSystem/Trigger.hpp"
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "Evolution/Actions/RunEventsAndDenseTriggers.hpp"
+#include "Evolution/EventsAndDenseTriggers/DenseTrigger.hpp"
+#include "Evolution/EventsAndDenseTriggers/EventsAndDenseTriggers.hpp"
+#include "Evolution/EventsAndDenseTriggers/Tags.hpp"
+#include "Framework/ActionTesting.hpp"
+#include "Framework/MockDistributedObject.hpp"
+#include "Framework/MockRuntimeSystem.hpp"
+#include "Framework/MockRuntimeSystemFreeFunctions.hpp"
+#include "Options/Protocols/FactoryCreation.hpp"
+#include "Parallel/Actions/SetupDataBox.hpp"
+#include "Parallel/PhaseDependentActionList.hpp"
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Event.hpp"
+#include "Time/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+template <typename Id>
+struct LinkedMessageId;
+namespace Parallel {
+template <typename Metavariables>
+class GlobalCache;
+}  // namespace Parallel
+
+namespace {
+template <typename ExpectedSystems>
+struct Submeasurement
+    : tt::ConformsTo<control_system::protocols::Submeasurement> {
+  template <typename ControlSystems>
+  using interpolation_target_tag = void;
+
+  using argument_tags = tmpl::list<>;
+
+  template <typename Metavariables, typename ParallelComponent,
+            typename ControlSystems>
+  static void apply(const LinkedMessageId<double>& /*measurement_id*/,
+                    Parallel::GlobalCache<Metavariables>& /*cache*/,
+                    const int& /*array_index*/,
+                    const ParallelComponent* /*meta*/,
+                    ControlSystems /*meta*/) {
+    // Test cases have either one or two systems, so this covers all
+    // permutations.
+    static_assert(
+        std::is_same_v<ControlSystems, ExpectedSystems> or
+        std::is_same_v<ControlSystems, tmpl::reverse<ExpectedSystems>>);
+    ++call_count;
+  }
+
+  static size_t call_count;
+};
+
+template <typename ExpectedSystems>
+size_t Submeasurement<ExpectedSystems>::call_count = 0;
+
+template <typename ExpectedSystems>
+struct Measurement : tt::ConformsTo<control_system::protocols::Measurement> {
+  using submeasurements = tmpl::list<Submeasurement<ExpectedSystems>>;
+};
+
+struct SystemB;
+
+struct SystemA : tt::ConformsTo<control_system::protocols::ControlSystem> {
+  static std::string name() { return "A"; }
+  using measurement = Measurement<tmpl::list<SystemA, SystemB>>;
+  struct process_measurement {};
+};
```

Comment:
I'm surprised this conforms to the protocol. Doesn't it also need a `using simple_tags = tmpl::list<>`?
Same for the others in this PR.

Reply:
Probably nothing used here does a check.  I've added some assertions.